provider/src/test/kotlin/org/gradle/kotlin/dsl/integration/GradleKotlinDslIntegrationTest.kt

```diff
@@ -766,6 +768,66 @@ class GradleKotlinDslIntegrationTest : AbstractIntegrationTest() {
             """))
     }
 
+    @Test
+    fun `can cross configure buildscript`() {
```

Comment:
:o: I wonder what this test is testing that it's not already covered by other tests and whether that couldn't be covered by a unit test instead.

Reply:
The very same test on `develop` fails so wasn't covered by other tests.

Init script compilation failure:
```
FAILURE: Build failed with an exception.

* Where:
Initialization script '/private/var/folders/mw/443dn1yd2g54gxchh0vtpqkr0000gn/T/junit4386537838000888481/can#20cross#20configure#20buildscript/some.init.gradle.kts' line: 3

* What went wrong:
Script compilation errors:

  Line 3:                 rootProject.buildscript {
                                                  ^ Type mismatch: inferred type is () -> ??? but Closure<(raw) Any!>! was expected
```

Settings script compilation failure:
```
FAILURE: Build failed with an exception.

* Where:
Settings file '/private/var/folders/mw/443dn1yd2g54gxchh0vtpqkr0000gn/T/junit5120365101693653083/can#20cross#20configure#20buildscript/settings.gradle.kts' line: 4

* What went wrong:
Script compilation errors:

  Line 4:                 rootProject.buildscript {
                                                  ^ Type mismatch: inferred type is () -> ??? but Closure<(raw) Any!>! was expected
```

And project build script silently fails to configure sub project classpath properly:
```
FAILURE: Build failed with an exception.

* Where:
Build file '/private/var/folders/mw/443dn1yd2g54gxchh0vtpqkr0000gn/T/junit3168091385484588138/can#20cross#20configure#20buildscript/sub/build.gradle.kts' line: 4

* What went wrong:
Script compilation error:

  Line 4:                     val deep = org.gradle.kotlin.dsl.fixtures.DeepThought()
                                                               ^ Unresolved reference: fixtures
```

I think it's fine as an integration test.

Reply:
> I think it's fine as an integration test.

Integration tests are fine and provide a lot of coverage but there's a qualitative difference between waiting 10+ seconds and waiting 50 milliseconds (not waiting at all) for a test to run. Let's try to err on the side of the latter.

Reply:
The integration test has value in that it assert that things are correct script-classpath-wise.
This is especially true for the third case described above that was silently failing before.